File: compiler/testData/loadJava/compiledJava/kotlinSignature/propagation/return/TwoSuperclassesMutableAndNot.java
package test;

import java.util.*;

public interface TwoSuperclassesMutableAndNot {

    public interface Super1 {
        Collection<String> foo();

        void dummy(); // to avoid loading as SAM interface
    }

    public interface Super2 {
        List<String> foo();

        void dummy(); // to avoid loading as SAM interface
    }

    public interface Sub extends Super1, Super2 {
        List<String> foo();
    }
}

File: kotlin-ide/idea/testData/compiler/loadJava/compiledKotlin/fromLoadJava/kotlinSignature/propagation/return/TwoSuperclassesMutableAndNot.kt
package test

public interface TwoSuperclassesMutableAndNot {

    public interface Super1 {
        public fun foo(): MutableCollection<String>

        public fun dummy() // to avoid loading as SAM interface
    }

    public interface Super2 {
        public fun foo(): List<String>

        public fun dummy() // to avoid loading as SAM interface
    }

    public interface Sub: Super1, Super2 {
        override fun foo(): MutableList<String>
    }
}
